doc/ref_impl/cntt-ri/chapters/chapter06.md

```diff
@@ -5,15 +5,25 @@
 
 ## Table of Contents
 * [6.1 Introduction](#6.1)
-* [6.2 Heading](#6.2)
-* [6.3 Heading](#6.3)
+* [6.2 OPNFV Pharos Specification](#6.2)
+* [6.3 Spine Leaf Topology](#6.3)
 
 <a name="6.1"></a>
 ## 6.1 Introduction
 
 <a name="6.2"></a>
-## 6.2 Heading
+## 6.2 OPNFV Pharos Specification
+
+The Reference Implemenation will follow the baseline OPNFV Pharos specification.  The following diagam illustrates the networking
+and servers required.
```

Comment:
@markbeierl Hi Mark, This is a great start. I'm not sure all of this has been fleshed out yet, other than in informal discussion, so we'll probably need to iterate a bit. Here are some initial comments (long):

- This RI is leveraging the original Pharos as it exists today, on an interim basis, and thus can get us started immediately. However, the actual target RI baseline is an enhanced version of Pharos, or, Pharos is being enhanced to align w/ RI. So, the way Sect 6.2 is currently written is correct if it's qualified that it's the new Pharos, as opposed to original Pharos. That said, AFAIK it's TBD what the enhanced Pharos will be called (2.0???).
- This RI is primarily intended to support and complement RC operations; it's a functional cross-section and minimal config, to allow testing. Except for an edge case, a telco is unlikely to deploy a 5 node (or even 15 node) instance, so we should ensure we set readers' expectations of the RI and diagram, appropriately.

Network related - Some of the following pertain to facets of the diagram inherited from the original Pharos (denoted by an asterisk), but I think we need to revisit these points and potentially revise. If I'm missing the Pharos intent of any of the items I listed below, if you, @trevgc or @lylavoie could advise, it would be appreciated.

1. The diagram, and these comments are tailored for a captive environment used to perform RC qualification. There are security and other issues that preclude this environment from being used in production. Hence, it should be clearly stated this diagram is NOT a production prescription. This is intended to set readers expectations, and prevent spending a lot of obvious discussion on why this diagram isn't suitable for prod, as shown.
1. There is no connection between the Admin subnet and the jump server // I'm assuming this is just an omission
1. Public Network*: There is a node labeled public network inline w/ the jump box // I'm guessing this is intended to indicate that connection goes to the public network; if so, it should be noted along the line, but not as an inline node
1. Jump Box Connectivity*: Assuming the above assumption is correct, the primary access to the jump box should be over a private subnet (i.e. NOT over the public internet, as implied by the previous upstream/inline node). This private network is extended to the remote user via the VPN GW.
1. Topology (Logical or Physical): The VPN GW will be instantiated on the jump box, but they're shown as 2 separate nodes (which I prefer, as they are independent functions), but that implies the topology diagram is logical. However, we're showing the physical servers, and other physical boxes, which implies the topology is physical. No doubt it's a combination of the two, so I'd propose we subtly indicate what is a logical function and what is a physical box.
1. Router Placement*: The router is shown after the VPN GW. Two points: 1) If we're going to show a router, I'd expect it to be before the VPN GW. 2) Assuming the router is in the intended location, what is it supposed to do there?
1. Internal Network Connectivity*: Why do we have all of the networks connected to the "router"? My understanding is remote access to these networks is provided via the jump box. The only reason I can think of for connecting these subnets to that router is if we intend for the VPN to directly expose/backhaul these networks to the remote user. If that's the intent, I'd recommend we use the jump box for that access and not have connections between those networks and the VPN and/or router. I'm concerned the routing will become a nightmare for users (and for lab ops), and would rather keep it simple. Worst case, a user that really needs that kind of access to resolve a particular issue, will know enough to be able to tunnel it over ssh through the jump box via the primary VPN.
1. Storage Network: We need to add a local storage network
1. Public Network (Tenant/VM-level Internet Access): All of the tenant VMs need the option of connecting to a network which has a GW to the public Internet, known as a "Provider Network" in the OpenStack world, such as for a DMZ or for simply downloading files.
1. Internet Access (Machine/Hypervisor-level): All of the machines, including the jump box, need access to the public Internet for downloading any needed programs, data, etc. This connection is strictly to aid in the maintenance of the node, and is NOT for exposing any services, nor any incoming remote access (i.e. it's pull, only). For a lab, this could be the same subnet used for the Public Network subnet, mentioned above.
1. Color Code & Internet Access Topology*: The overall color code is good, however, it seems to get a little loose near the VPN GW. I like red for Public Internet, so the Internet Access subnet going to all nodes, should be red. However, that network needs a GW connection to a router north of the VPN GW, so as mentioned above, we need a router between the Firewall and the VPN GW to tie the Internet Access and Provider Network to. The current diagram shows the internet access coming from south of the VPN GW, which doesn't make sense in this context, as there shouldn't be any Internet access coming through the VPN.
1. Color Code*: Need to add a color for the subnet that's routeable over the VPN to the remote client
1. VLANs: While we can't specify absolute VIDs, I propose we specify relative/offset VIDs. For example, offset VID 0 could be Lights Out, 1 could be Admin, etc. The offsets would then be added to a base VID assigned to the POD (specified by the site networking person, not by CNTT). For example, if the site assigned POD 1 a base VID of 100, then the actual LO VID would be 100 (100(base)+0(LO offset)), Admin would be 101 (100 + 1), etc. This is primarily intended to clarify the diagram, making it usable in practice and help it communicate its intent, as opposed to simply being prescriptive.

Feel free to reach out, if you'd like to discuss.
Thanks,
-Mark

Reply:
Thanks, @markshostak.  I do have some questions:

- There are two issues: #405 and #406.  Is #405 supposed to be for the generic case and #406 an attempt to document what OPNFV can provide right now?
- How do we want to deal with the multi-rack production scale RI documentation?  I would guess that HA is going to be part of the certification?
- Agreed that this pass of the document was first to replicate what was originally created in the old OPNFV wiki and is lacking important details.  I would like to keep this logical, where we show connections to the various network segments (VLANs, etc) but then how do we want to indicate that there should be two switches for HA?  Or is that not in the scope for this document?
- I'd prefer to get rid of the VPN and such and just focus on the jump/foundation node towards the rest of the servers.

Reply:
Hi Mark,
>   There are two issues: #405 and #406. Is #405 supposed to be for the generic case and #406 an attempt to document what OPNFV can provide right now?

That's how I read it, but it's not cut in stone; we have latitude. 

>    How do we want to deal with the multi-rack production scale RI documentation? I would guess that HA is going to be part of the certification?

Negative. Not at this time. 

>    Agreed that this pass of the document was first to replicate what was originally created in the old OPNFV wiki and is lacking important details. I would like to keep this logical, where we show connections to the various network segments (VLANs, etc) but then how do we want to indicate that there should be two switches for HA? Or is that not in the scope for this document?

Easy! You nailed it - not in scope.

>   I'd prefer to get rid of the VPN and such and just focus on the jump/foundation node towards the rest of the servers.

My 2 cents - I think the VPN is important, as when I was trying to figure how this worked, that was a key data point. Before that it was, am I supposed to SSH into it, or how do I get there???

HTH
-Mark